src/main/java/ca/wolfram/beta/symath/operations/MultiplyNode.java

```diff
@@ -24,18 +34,61 @@ public static MathNode create(MathNode... nodes) {
 
     @Override
     public boolean simplify() {
-        // TODO
+        long constant = simplify(0, 1L);
+        if (constant != 1L)
+            getChildren().add(BaseNode.create(constant));
         return super.simplify();
     }
 
+    /**
+     * Helper method for simplification
+     * Starts the listIterator at index, with history of constant, and continues from there
+     * Extracts all constant integers and multiplies values to {@code constant}
+     * If child is of type MULTIPLY, add its respective children to the current children and recursively simplify at the new index
+     *
+     * @param index    children index to start iteration
+     * @param constant integer summation of child nodes before index
+     * @return new integer product
+     */
+    private long simplify(int index, long constant) {
+        List<MathNode> childrenToRaise = new ArrayList<>();
+        final ListIterator<MathNode> each = getChildren().listIterator(index);
+        while (each.hasNext()) {
+            MathNode next = each.next();
+            if (next.getType() == NodeType.MULTIPLY) {
+                childrenToRaise.addAll(next.getChildren());
+                each.remove();
+                continue;
+            }
+            if (MathUtils.isReciprocal(next) && MathUtils.isConstantInt(next.getChildren().get(0))) {
+                double potential = constant / next.getChildren().get(0).eval(null);  // reduce precision loss
+                if (potential == (long) potential) {
```

Comment:
Is this actually the best way to do this? and does it actually work? Maybe we can add this function to MathUtils and just fix it in one place in the future.